src/kernel.ts

```diff
@@ -1023,6 +1032,7 @@ class Kernel implements IKernel {
   private _commPromises: Map<string, Promise<IComm>> = null;
   private _comms: Map<string, IComm> = null;
   private _isWebSocketClosing = false;
+  private _targetRegistry: { [key: string]: (comm: IComm, msg: IKernelMessage) => void; } = Object.create(null);
```

Comment:
Cool, I didn't know about `Object.create(null)`.  Any reason we're not using a Map here?